### PR TITLE
Cli tool

### PR DESCRIPTION
**Pull request by CarolineMorton**

### CODE REVIEW

File: utils_entrypoint.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+
+
+def load_default_args(match_config):
```
Comment (bloodearnest) [inline utils_entrypoint.py:3]: A common pattern for this kind of defaults config is to creatre a defaults dict and then overwrite it with use config.

e.g.

```python

DEFAULTS = {
        "closest_match_variable":  None,
        "date_exclusion_variables": None,
        "min_matches_per_case": 0,
        "replace_match_index_date_with_case": None,
        "output_suffix": "",
        "indicator_variable_name": "case",
        "output_path": "tests/test_output",
        "input_path": "tests/test_data",
        "drop_cases_from_matches": False,
}

def load_config(match_config):
    cfg = DEFAULTS.copy()
    cfg.update(match_config)
    return cfg

```

File: utils_entrypoint.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        match_config["replace_match_index_date_with_case"] = None
+
+    if "output_suffix" not in match_config:
+        match_config["output_suffix"] = ""
+
+    if "indicator_variable_name" not in match_config:
+        match_config["indicator_variable_name"] = "case"
+
+    if "output_path" not in match_config:
+        match_config["output_path"] = "tests/test_output"
```
Comment (bloodearnest) [inline utils_entrypoint.py:25]: I'm wary of testing defaults. I'd be inclined to either have no default, or make it "outputs/something"

File: tests/test_json/test_json_3.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+  "inputs": {
+    "cases": "input_cases.csv",
```
Comment (bloodearnest) [inline tests/test_json/test_json_3.json:3]: Ok, so the original plan was to have `inputs` generated automatically by the job runner.

If we've got to manually tag specific inputs, we may need to revisit this.

**Review by bloodearnest — APPROVED** [review]
LGTM.

Left a couple of suggestions but not blocking

File: tests/test_json/test_json_3.json (hunk 3 above)
Comment (alexwalkercebm) [reply tests/test_json/test_json_3.json:3]: There are occasions when you want to do something _between_ the job runner extract and the matching, so we at least need an option to be able to manually specify inputs.

**Review by alexwalkercebm — APPROVED** [review]

File: utils_entrypoint.py (hunk 2 above)
Comment (CarolineMorton) [reply utils_entrypoint.py:25]: noted - and changed

**Pull request merged**